daml-lf/spec/daml-lf-1.rst

```diff
@@ -2087,84 +2103,98 @@ Int64 functions
-  value to be rounded is exactly half-way between two multiples,
-  rounds toward the even one, following the `banker's rounding
-  convention
-  <https://en.wikipedia.org/wiki/Rounding#Round_half_to_even>`_. Throws
-  an exception if the integer is not between -27 and 10 inclusive.
+  Rounds the decimal to the closest multiple of ``10ⁱ`` where ``i`` is
+  integer argument.  In case the value to be rounded is exactly
+  half-way between two multiples, rounds toward the even one,
+  following the `banker's rounding convention
+  <https://en.wikipedia.org/wiki/Rounding#Round_half_to_even>`_.  The
+  scale of the inputs and the ouput is given by the type parameter
+  `α`.  Throws an exception if the integer is not between `α-37` and
+  `α` inclusive.
 
-* ``LESS_EQ_DECIMAL : 'Decimal' → 'Decimal' → 'Bool'``
+* ``LESS_EQ_NUMERIC : ∀ (α : nat) . 'Numeric' α → 'Numeric' α → 'Bool'``
 
   Returns ``'True'`` if the first decimal is less or equal than the
-  second, ``'False'`` otherwise.
+  second, ``'False'`` otherwise.  The scale of the inputs is given by
+  the type parameter `α`.
 
-* ``GREATER_EQ_DECIMAL : 'Decimal' → 'Decimal' → 'Bool'``
+* ``GREATER_EQ_NUMERIC : ∀ (α : nat) . 'Numeric' α → 'Numeric' α → 'Bool'``
 
-  Returns ``'True'`` if the first decimal is greater or equal than
-  the second, ``'False'`` otherwise.
+  Returns ``'True'`` if the first decimal is greater or equal than the
+  second, ``'False'`` otherwise. The scale of the inputs is given by
+  the type parameter `α`.
 
-* ``LESS_DECIMAL : 'Decimal' → 'Decimal' → 'Bool'``
+* ``LESS_NUMERIC : ∀ (α : nat) . 'Numeric' α → 'Numeric' α → 'Bool'``
 
   Returns ``'True'`` if the first decimal is strictly less than the
-  second, ``'False'`` otherwise.
+  second, ``'False'`` otherwise.  The scale of the inputs is given by
+  the type parameter `α`.
 
-* ``GREATER_DECIMAL : 'Decimal' → 'Decimal' → 'Bool'``
 
-  Returns ``'True'`` if the first decimal is strictly greater than
-  the second, ``'False'`` otherwise.
+* ``GREATER_NUMERIC : ∀ (α : nat) . 'Numeric' α → 'Numeric' α → 'Bool'``
+
+  Returns ``'True'`` if the first decimal is strictly greater than the
+  second, ``'False'`` otherwise.  The scale of the inputs is given by
+  the type parameter `α`.
 
-* ``EQUAL_DECIMAL : 'Decimal' → 'Decimal' → 'Bool'``
+* ``EQUAL_NUMERIC : ∀ (α : nat) . 'Numeric' α → 'Numeric' α → 'Bool'``
 
   Returns ``'True'`` if the first decimal is equal to the second,
-  ``'False'`` otherwise.
+  ``'False'`` otherwise.  The scale of the inputs is given by the type
+  parameter `α`.
 
-* ``TO_TEXT_DECIMAL : 'Decimal' → 'Text'``
+* ``TO_TEXT_NUMERIC : ∀ (α : nat) . 'Numeric' α → 'Text'``
 
-  Returns the decimal string representation of the decimal.
+  Returns the decimal string representation of the decimal.  The scale
+  of the input is given by the type parameter `α`.
 
-* ``FROM_TEXT_DECIMAL : 'Text' → 'Optional' 'DECIMAL'``
+* ``FROM_TEXT_NUMERIC : ∀ (α : nat) .'Text' → 'Optional' 'Numeric' α``
```

Comment:
Is this a sufficient specification to include that the engine implementation of this primitive requires `α` to be reified? Consider

```hs
parseDecimal : Text -> Optional (Numeric n)
parseDecimal = primitive @"BEDecimalFromText"

parseDecimals : [Text] -> [Optional (Numeric n)]
parseDecimals = fmap parseDecimal

fives : [Optional (Numeric 5)]
fives = parseDecimals ["5"]